aclk/agent_cloud_link.c

```diff
@@ -133,6 +141,41 @@ struct aclk_query_queue {
     u_int64_t count;
 } aclk_queue = { .aclk_query_head = NULL, .aclk_query_tail = NULL, .count = 0 };
 
+/*
+ * After a connection failure -- delay in milliseconds
+ * When a connection is established, the delay function
+ * should be called with
+ *
+ * mode 0 to reset the delay
+ * mode 1 to sleep for the calculated amount of time [0 .. ACLK_MAX_BACKOFF_DELAY * 1000] ms
+ *
+ */
+unsigned long int aclk_reconnect_delay(int mode)
+{
+    static  int fail = -1;
+    unsigned long int delay;
+
+    if (!mode || fail == -1) {
+        srandom(time(NULL));
+        fail = mode-1;
+        return 0;
+    }
+
+    delay = (1 << fail);
```

Comment:
I am studying the code from top to bottom, so it is possible that my answer is below, but case it is not, I will do a question. Is there any possibility of `fail` to be bigger than 64? Case it has this possibility, we need to bring an initial test here to avoid overflow.